server/src/main/java/org/opensearch/action/tiering/RestWarmTieringAction.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.tiering;
+
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.RestHandler;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.action.RestToXContentListener;
+
+import java.util.List;
+
+import static java.util.Collections.singletonList;
+import static org.opensearch.rest.RestRequest.Method.POST;
+
+/**
+ * Rest Tiering API class to move index from hot to warm
+ *
+ * @opensearch.experimental
+ */
+public class RestWarmTieringAction extends BaseRestHandler {
+
+    @Override
+    public List<RestHandler.Route> routes() {
+        return singletonList(new RestHandler.Route(POST, "/{index}/_tier/{tier}"));
+    }
```

Comment:
I'm not sure if we have API standards/conventions somewhere but should the `tier` parameter actually be in the request body instead? My (admittedly pretty limited) understanding is that typically we would only have resource identifiers in the URL parameters while anything related to *how* the resources are modified should go in the request body.

Also this probably needs to get added here: https://github.com/opensearch-project/opensearch-api-specification?tab=readme-ov-file

Reply:
sure i will it in the specification as well.

The idea of keeping the tier option as body was also discussed, however we wanted to avoid having the body to make it easier for the customers to call the tiering api without having a necessary body.
Also, this API follows a model of performing an action(tier=hot/warm) on the resource(index) which most of the opensearch apis follow.

Reply:
For reference: https://github.com/opensearch-project/OpenSearch/issues/14456

Reply:
raised a pr here: https://github.com/opensearch-project/opensearch-api-specification/pull/368